### ISSUE TITLE
 Dialogs miss max length validation for text entered by users 

### ISSUE THREAD

**Issue by jswiderski** [ckfinder/ckfinder#386]
1. Open CKFinder
2. Click on any folder.
3. Click on New Subfolder button and enter a very long name.

**Actual Result: **
Error will be throws

**Expected Result:**
Validation dialog should be shown, informing user that maximum allowed length has been exceeded.

Other Information:
The same can be reproduced for Rename File dialog and Rename File Dialog in Image Editor.

We could perhaps make that MAX_LENGHT value configurable so that different developers could set different values.

**Comment by jswiderski** [comment]
Temporary Workaround:

```
CKFinder.widget( 'ckfinder-widget', {
  language : 'en',
  onInit: function( finder ) {
    finder.on( 'dialog:CreateFolder:ok', function( evt ) {       
      const MAX_LENGTH = 20;
      var folderName = evt.data.view.model.get('folderName');
 
      if( folderName.length > MAX_LENGTH ) {    
          finder.request( 'dialog:info', {
            msg: 'Error: Folder Name cannot be longer than ' + MAX_LENGTH + ' characters.',
            title: 'Error Dialog'
          });
          evt.cancel();
      }             
    });

  }
});
```

**Comment by SpencerGresoro** [comment]
> Temporary Workaround:
> 
> ```
> CKFinder.widget( 'ckfinder-widget', {
>   language : 'en',
>   onInit: function( finder ) {
>     finder.on( 'dialog:CreateFolder:ok', function( evt ) {       
>       const MAX_LENGTH = 20;
>       var folderName = evt.data.view.model.get('folderName');
>  
>       if( folderName.length > MAX_LENGTH ) {    
>           finder.request( 'dialog:info', {
>             msg: 'Error: Folder Name cannot be longer than ' + MAX_LENGTH + ' characters.',
>             title: 'Error Dialog'
>           });
>           evt.cancel();
>       }             
>     });
> 
>   }
> });
> ```

This will not work, it needs to account for the length of the entire path. 

Create a folder with 156 characters

Now Create a sub directory of this folder, . . . :(

It's throwing BOTH

Unknown error

and in some cases

A splash screen error.

I don't know enough about the finder to comment on why this might be happening, but it chokes up in these situations.

**Comment by jswiderski** [comment]
@SpencerGresoro I don't think it is a good idea to go for the maximum since user might use Latin and non-Latin characters which will take up more space because of encoding.  As a developer you have the power to decide what the MAX_LENGTH will be and in my personal opinion you should go for something safe like 60 or 70 characters.

We will try to find a solution here however the maximum may not be avoidable (we will see about that).

**Comment by jswiderski** [comment]
It seems this problem is more problematic then first assumed:

> 
> I can create a folder at 100 characters, then create a subfolder at another 100 characters and get the break. And it’s not consistent.
> Setting the max length low will solve a lot of the issues, but not all when nesting folders with long names and create a large path that way.

**Comment by SpencerGresoro** [comment]
> It seems this problem is more problematic then first assumed:
> 
> > I can create a folder at 100 characters, then create a subfolder at another 100 characters and get the break. And it’s not consistent.
> > Setting the max length low will solve a lot of the issues, but not all when nesting folders with long names and create a large path that way.

I think a good start would be too tie into whatever is generating that 'Unknown Error' modal popup on commit (create folder).

Maybe something is wrapped in a try/catch, :)
